airbyte-cdk/python/airbyte_cdk/sources/declarative/spec/spec.py

```diff
@@ -29,14 +29,14 @@ def generate_spec(self) -> ConnectorSpecification:
         Returns the connector specification according the spec block defined in the low code connector manifest.
         """
 
-        obj: dict[str, Mapping[str, Any] | str | AuthFlow] = {"connectionSpecification": self.connection_specification}
+        obj: dict[str, Mapping[str, Any] | str | AdvancedAuth] = {"connectionSpecification": self.connection_specification}
 
         if self.documentation_url:
             obj["documentationUrl"] = self.documentation_url
         if self.advanced_auth:
-            obj["advanced_auth"] = self.advanced_auth
-            # Get enum value
-            obj["advanced_auth"].auth_flow_type = obj["advanced_auth"].auth_flow_type.value  # type: ignore # We know this is always assigned to an AuthFlow which has the auth_flow_type field
+            self.advanced_auth.auth_flow_type = self.advanced_auth.auth_flow_type.value  # type: ignore # We know this is always assigned to an AuthFlow which has the auth_flow_type field
+            # Map CDK AuthFlow model to protocol AdvancedAuth model
+            obj["advanced_auth"] = AdvancedAuth.parse_obj(self.advanced_auth.dict())
```

Comment:
Change: The `AuthFlow` CDK model and the `AdvancedAuth` protocol model have all the same values, but are different models. In pydantic v1, we could set `obj["advanced_auth"] = self.advanced_auth` with these different models no problem. In pydantic V2, we need to explicitly turn it into an `AdvancedAuth` since that is the model type that `ConnectorSpecification` is expecting. 

Likely related to 
<img width="719" alt="image" src="https://github.com/airbytehq/airbyte/assets/22333512/27734612-61fe-49e8-a3bd-365a0d9e796e">
but not exactly sure